### PR TITLE
[release-4.15] OCPBUGS-36586: Add config map hooks

### PR DESCRIPTION
**Pull request by stephenfin**
This is a manual backport of #174

### CODE REVIEW

File: pkg/operator/starter.go [hunk 1]
```diff
@@ -211,5 +305,5 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
 
 	<-ctx.Done()
 
-	return fmt.Errorf("stopped")
+	return nil
```
Comment (mandre) [inline pkg/operator/starter.go:308]: It seems this last chunk is extra (part of https://github.com/openshift/openstack-cinder-csi-driver-operator/pull/155 that was not backported to 4.15).

Comment (stephenfin) [reply pkg/operator/starter.go]: Done

**Review by mandre — COMMENTED** [review]
/lgtm
/approve
/label backport-risk-assessed

**Pull request merged**